### ISSUE TITLE
Make it possible to run forked version of a docs site w/o having to edit siteConfig.js

### ISSUE THREAD

**Issue by brooklynrob** [finos/open-developer-platform#181]
Currently to run an instance of a docs site on one's fork - which can be useful during doc development, editing, and review to look at changes before they end up in production - the siteConfig.js file, specifically the `cname`, `url`, `baseUrl`, and `organizationName` values, have to be edited/overridden.

![image](https://user-images.githubusercontent.com/2136904/95351263-07d9ec80-088f-11eb-8b58-df28228b7a48.png)

This is somewhat risky as if a developer inadvertently includes siteConfig.js with their changes to overwrite these values in a commit, and in turn a PR, to the upstream repo, and that PR is accepted, the upstream build could fail/break (e.g., if a developer used, say, `git add` instead of `git add <filename>` and inadvertently pulled siteConfig.js into a commit)

We should make it possible to run a forked version w/o having to do this.

Perhaps one way to do this would be to make use of environment variables and/or an include file that could be edited outside of the siteConfig.js file.

**Comment by adamj-codethink** [comment]
Good idea, maybe something like: https://codeburst.io/node-js-best-practices-smarter-ways-to-manage-config-files-and-variables-893eef56cbef ?

**Comment by maoo** [comment]
This should be already available, see the `Patch forked version` step on https://github.com/finos/open-developer-platform/blob/master/.github/workflows/docusaurus.yml#L41-L49

To test it:
- fork out this repo
- enable the github action on the fork
- make a change in the `docs` folder
- see if the patch works as expected (see the changes reported by rob in the issue description), by checking `siteConfig.js` in the `gh-pages` branch

@adamj-codethink - feel free to give it a try!

In the future, I think it would be worth trying Netlify and see if/how much it can help us.

Thank you!

**Comment by adamj-codethink** [comment]
Awesome thanks @maoo, i will give this a whirl :D

**Comment by adamj-codethink** [comment]
@maoo I spent some time playing with this, but i think the rename from alloy -> legend may have broke the action?> https://github.com/finos/legend/pull/275

**Comment by maoo** [comment]
TY, that's awesome! I've reviewed/approved the PR. I wonder if that's the only issue you found, and whether the docs on https://odp.finos.org/docs/development-infrastructure/project-documentation are correct.

**Comment by adamj-codethink** [comment]
I posted some comments on the legend PR @maoo :)

**Comment by adamj-codethink** [comment]
This works for me now finos/legend#275 has been merged
